src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -374,8 +374,8 @@ function MoneyRequestConfirmationList({
         const taxAmount = getTaxAmount(transaction, policy).toString();
         const amountInSmallestCurrencyUnits = CurrencyUtils.convertToBackendAmount(Number.parseFloat(taxAmount));
 
-        if (transaction?.taxAmount && previousTransactionAmount === transaction?.amount && previousTransactionCurrency === transaction?.currency) {
-            return IOU.setMoneyRequestTaxAmount(transaction?.transactionID, transaction?.taxAmount, true);
+        if (transaction?.taxAmount !== null && previousTransactionAmount === transaction?.amount && previousTransactionCurrency === transaction?.currency) {
+            return IOU.setMoneyRequestTaxAmount(transactionID, transaction?.taxAmount ?? 0, true);
```

Comment:
I think we need to revert the change in this file: src/components/MoneyRequestConfirmationList.tsx

I am not sure why we need to apply this change but It causes this bug:

The Tax amount is 0 when creating a new money request

https://github.com/Expensify/App/assets/141406735/8161487f-90e4-4aab-9eb5-2484d7ba5856

Reply:
I think this is because of currency. Could you try with another currencies like GBP, EUR, USD and such

Reply:
> I am not sure why we need to apply this change but It causes this bug:

transaction?.taxAmount check is wrong because if taxAmount is zero, it doesn't work

Reply:
Saw similar bug here https://github.com/Expensify/App/pull/40443#issuecomment-2106818414

Reply:
https://github.com/Expensify/App/assets/141406735/f8fe14e6-b3f9-4374-a8e3-c5e14a2ac432

This bug still reproduces with other currencies, I checked the RCA and the reason is that we added the null check in  this PR

Reply:
https://github.com/Expensify/App/blob/5c619c7d244c0f94aec962258c57a922c01fecbd/src/components/MoneyRequestConfirmationList.tsx#L377-L381

In this case, transaction?.taxAmount is 0 so we need to use amountInSmallestCurrencyUnits to update taxAmount

Reply:
I'll check this

Reply:
Okay, I'll revert for now, but I'm pretty sure this is responsible for an undiscovered bug

Reply:
done, and thank you for thorough test!